src/infiniteGrid.js

```diff
@@ -335,8 +336,9 @@ eg.module("infiniteGrid", ["jQuery", eg, window, document, "Outlayer"], function
 					});
 				}
 			} else {
-				if (this.isRecycling() && this._removedContent > 0 &&
-					(ele = this._topElement || this.getTopElement())) {
+				if (this.isRecycling() && this._removedContent > 0) {
+					ele = $.isEmptyObject(this._topElement) ?
+						this.getTopElement() : this._topElement;
```

Comment:
@sculove Thank you for your kind explanation.

And I suggest... How about to cache `this._bottomElement` or `this._top_Element` if they are empty?

for example
```js
if ($.isEmptyObject(this._bottomElement)) {
  this._bottomElement = this.getBottomElement();
}
ele = this._bottomElement;
```